### ISSUE TITLE
"Correct Indentation" results in ClassCastException in log

### ISSUE THREAD

**Issue by davidmichaelkarr** [groovy/groovy-eclipse#231]
Using Neon.1a & 2.9.2.xx-201611160028-e46.

I associated "Jenkinsfile" with the groovy editor.  When I select the entire file and press Ctrl+i, nothing happens, but I see this in the log:

> Caused by: java.lang.ClassCastException: org.eclipse.jdt.internal.core.CompilationUnit cannot be cast to org.codehaus.jdt.groovy.model.GroovyCompilationUnit
	at org.codehaus.groovy.eclipse.refactoring.actions.FormatGroovyAction.run(FormatGroovyAction.java:70)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.dispatchRun(SelectionDispatchAction.java:275)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.run(SelectionDispatchAction.java:249)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:473)
	at org.eclipse.jface.commands.ActionHandler.execute(ActionHandler.java:118)
	... 55 more

**Comment by eric-milles** [comment]
Ugh, looks like the file did not get considered as groovy by the compiler.  Can you send your steps for mapping a file type to the groovy editor so I can take a look at it?

**Comment by davidmichaelkarr** [comment]
I went into Preferences, then "File Associations", underneath "General"->"Editors". I clicked "Add", entered "Jenkinsfile" for the entire file type", then selected "Groovy Editor" for the associated editor.

In the editor view, the tab icon has a lowercase "g" icon, and in the code area, I see keywords and some values in violet/purple. The right-click context menu has a "Groovy refactor" submenu.
